src/exchange/account/home/account-home.component.js

```diff
@@ -276,17 +328,46 @@
         return this.$translate.instant('exchange_tab_ACCOUNTS_state_UNKNOWN');
       }
 
-      let formattedAccounts = _(accounts)
-        .get('list.results', [])
-        .map(account => _(account)
-          .assign({
+      function filterCompany(account) {
+        const companyValue = account.company || '';
+
+        return _
+          .filter(criteria, { property: 'company' })
+          .every((criterium) => {
+            switch (criterium.operator) {
+              case 'contains':
+                return companyValue.includes(criterium.value);
+              case 'containsNot':
+                return !companyValue.includes(criterium.value);
+              case 'startsWith':
+                return _.startsWith(companyValue, criterium.value);
+              case 'endsWith':
+                return _.endsWith(companyValue, criterium.value);
+              case 'is':
+                return companyValue === criterium.value;
+              case 'isNot':
+                return companyValue !== criterium.value;
+              default:
+                return true;
+            }
+          });
+      }
+
+      let formattedAccounts = _.get(accounts, 'list.results', [])
+        .map(account => ({
+          ...account,
+          ...{
```

Comment:
```suggestion

```
You could just write directly the propeties and their values ?

Reply:
The whole 50 ?